apis/resourcecontrollerv2/v1alpha1/resourceinstance_types.go

```diff
@@ -38,7 +34,8 @@ type ResourceInstanceParameters struct {
 
 	// The name of the resource group where the instance is deployed
 	// +immutable
-	ResourceGroupName string `json:"resourceGroupName"`
+	// +optional
+	ResourceGroupName string `json:"resourceGroupName,omitempty"`
```

Comment:
Here I've intentionally made the `ResourceGroupName` optional.  This is a trick we do in the Terraform generator.  It is relatively easy to default this to the "Default" resource group for the user, so it's just nicer to make it optional.

Reply:
good point, I agree.